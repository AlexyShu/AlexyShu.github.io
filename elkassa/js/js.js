const popup = document.querySelector(".popup");
const callBtn = document.querySelector(".header-callback");
const requestBtn = document.querySelector(".request");
const closeBtn = document.querySelector(".btn-close");

const kioskPopup = document.querySelector(".kiosk-popup");
const kioskRequestBtn = document.querySelector(".kiosk-request");
const closeKioskBtn = document.querySelector(".btn-kiosk-close");

const infoPopup = document.querySelector(".info-popup");
const infoRequestBtn = document.querySelector(".info-request");
const closeInfoBtn = document.querySelector(".btn-info-close");

const contactsPopup = document.querySelector(".contacts-popup");
const contactsRequestBtn = document.querySelector(".contacts-request");
const closeContactsBtn = document.querySelector(".btn-contacts-close");

const pricePopup = document.querySelector(".price-popup");
const priceRequestBtnFirst = document.querySelector(".price-request_first");
const priceRequestBtnSecond = document.querySelector(".price-request_second");
const priceRequestBtnThird = document.querySelector(".price-request_third");
const closePriceBtn = document.querySelector(".btn-price-close");

function closePricePopup() {
	pricePopup.classList.add("visually-hidden");
	pricePopup.classList.remove("modal-show-animation"); 
};

function showPricePopup() {
	pricePopup.classList.remove("visually-hidden"); 
	pricePopup.classList.add("modal-show-animation");
};

function closeContactsPopup() {
	contactsPopup.classList.add("visually-hidden");
	contactsPopup.classList.remove("modal-show-animation"); 
};

function showContactsPopup() {
	contactsPopup.classList.remove("visually-hidden"); 
	contactsPopup.classList.add("modal-show-animation");
};

function closeInfoPopup() {
	infoPopup.classList.add("visually-hidden");
	infoPopup.classList.remove("modal-show-animation"); 
};

function showInfoPopup() {
	infoPopup.classList.remove("visually-hidden"); 
	infoPopup.classList.add("modal-show-animation");
};

function closeKioskPopup() {
	kioskPopup.classList.add("visually-hidden");
	kioskPopup.classList.remove("modal-show-animation"); 
};

function showKioskPopup() {
	kioskPopup.classList.remove("visually-hidden"); 
	kioskPopup.classList.add("modal-show-animation");
};

function closePopup() {
	popup.classList.add("visually-hidden");
	popup.classList.remove("modal-show-animation"); 
};

function showPopup() {
	popup.classList.remove("visually-hidden"); 
	popup.classList.add("modal-show-animation");
};

callBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	showPopup(); 
});

requestBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	showPopup(); 
});

closeBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	closePopup(); 
});

kioskRequestBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	showKioskPopup(); 
});

closeKioskBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	closeKioskPopup(); 
});

infoRequestBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	showInfoPopup(); 
});

closeInfoBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	closeInfoPopup(); 
});

contactsRequestBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	showContactsPopup(); 
});

closeContactsBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	closeContactsPopup(); 
});

priceRequestBtnFirst.addEventListener("click", function(evt) {
	evt.preventDefault();
	showPricePopup(); 
});

priceRequestBtnSecond.addEventListener("click", function(evt) {
	evt.preventDefault();
	showPricePopup(); 
});

priceRequestBtnThird.addEventListener("click", function(evt) {
	evt.preventDefault();
	showPricePopup(); 
});

closePriceBtn.addEventListener("click", function(evt) {
	evt.preventDefault();
	closePricePopup(); 
});