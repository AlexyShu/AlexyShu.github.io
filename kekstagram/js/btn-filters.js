'use strict';

(function () {

  var BTN_CLASS = 'img-filters__button';
  var imgFilters = document.querySelector('.img-filters');
  var filterForm = document.querySelector('.img-filters__form');
  var photos = [];

  var showFilters = function (pics) {
    photos = pics;
    imgFilters.classList.remove('img-filters--inactive');
  };

  filterForm.addEventListener('click', function (evt) {
    var target = evt.target;
    if (target.classList.contains(BTN_CLASS)) {
      var activeBtn = filterForm.querySelector('.' + BTN_CLASS + '--active');
      activeBtn.classList.remove(BTN_CLASS + '--active');
      target.classList.add(BTN_CLASS + '--active');
      debouncedFilter(target.id);
    }
  });

  var filterPhotos = function (filterId) {
    var filteredPhotos = [];
    switch (filterId) {
      case 'filter-popular':
        filteredPhotos = photos;
        break;
      case 'filter-random':
        filteredPhotos = shuffle(photos.slice());
        break;
      case 'filter-discussed':
        filteredPhotos = photos.slice().sort(function (a, b) {
          return b.comments.length - a.comments.length;
        });
        break;
    }
    window.renderPic.renderPhotos(filteredPhotos);
  };

  var debouncedFilter = window.debounce.removeDebounce(filterPhotos);

  var shuffle = function (pics) {
    var elements = [];
    for (var i = 0; i < 10; i++) {
      var randomIndex = window.util.getRandomNumber(0, pics.length - 1);
      var randomItem = pics.splice(randomIndex, 1)[0];
      elements.push(randomItem);
    }
    return elements;
  };

  window.btnFilters = {
    showFilters: showFilters
  };

})();
