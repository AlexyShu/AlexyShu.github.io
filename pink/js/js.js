document.querySelector("body").classList.remove("no-js");var btn_off=document.querySelector(".page-header__taggle-off"),btn_on=document.querySelector(".page-header__taggle-on"),popup=document.querySelector(".page-header");btn_on.addEventListener("click",function(e){popup.classList.add("menu-active"),btn_on.classList.add("page-header__taggle-on--hidden"),btn_off.classList.add("page-header__taggle-off--show")}),btn_off.addEventListener("click",function(e){popup.classList.remove("menu-active"),btn_off.classList.remove("page-header__taggle-off--show"),btn_on.classList.remove("page-header__taggle-on--hidden")});