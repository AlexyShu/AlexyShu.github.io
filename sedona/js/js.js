var btn = document.querySelector(".hotels-search");
var popup = document.querySelector(".hotels-search-form");
var data1 = document.querySelector(".appointment-item-date1");
var data2 = document.querySelector(".appointment-item-date2");


btn.addEventListener("click", function(evt) {
	evt.preventDefault();
	popup.classList.toggle("modal-show");
	popup.classList.add("modal-show-animation");  
});

popup.addEventListener("animationend", function() {
	popup.classList.remove("modal-show-animation");
	popup.classList.remove("modal-error");
});

popup.addEventListener("submit", function(evt) {
	if (!data1.value || !data2.value) { 
	    evt.preventDefault(); 
		popup.classList.add("modal-error");  	
   }
});