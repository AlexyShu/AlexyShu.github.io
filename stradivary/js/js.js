
// кнопка вверх
const btnScroll = document.querySelector(".scroll-to-top");

window.addEventListener('scroll', function() {
    if (pageYOffset > 400) {
		btnScroll.classList.remove("visually-hidden"); 
	} if (pageYOffset < 400) {
		btnScroll.classList.add("visually-hidden"); 
	}
});