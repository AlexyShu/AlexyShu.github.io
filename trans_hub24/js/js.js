'use strict';

const navigation = document.querySelector('.nav-info');
const mainNavigation = document.querySelector('.nav-main');
const mainNavigationLink = document.querySelectorAll('.nav-list-link');
const navString = document.querySelector('.nav-info_phone');
const navLink = document.querySelectorAll('.nav-info_list-link');
const entryLink = document.querySelector('.nav-info_entry-link');
const registrationLink = document.querySelector('.nav-info_registration-link');
const logo = document.querySelector('.nav-list_logo');
const logoScroll = document.querySelector('.nav-list_logo-hidden');
const openMenuBtn = document.querySelector('.open-menu_btn');
const closeMenuBtn = document.querySelector('.close-menu_btn');
const littleDisplayNav = document.querySelector('.header_nav-little-display');

window.addEventListener('scroll', function() {
    if (pageYOffset > 1073) {
        navigation.style.background = '#ffffff';
        mainNavigation.style.background = '#ffffff';
        navString.style.color = '#B9B9B9';
        for (let i = 0; i < navLink.length; i++) {
            navLink[i].style.color = '#B9B9B9';
        }
        entryLink.style.color = '#B9B9B9';
        registrationLink.style.color = '#B9B9B9';
        for (let y = 0; y < navLink.length; y++) {
            mainNavigationLink[y].style.color = '#004460';
        }
        logo.style.display='none';
        logoScroll.style.display='block';
    } else {
        navigation.style.background = 'transparent';
        mainNavigation.style.background = 'transparent';
        navString.style.color = 'rgba(255, 255, 255, 0.5);';
        for (let i = 0; i < navLink.length; i++) {
            navLink[i].style.color = 'rgba(255, 255, 255, 0.5);';
        }
        entryLink.style.color = 'rgba(255, 255, 255, 0.5);';
        registrationLink.style.color = 'rgba(255, 255, 255, 0.5);';
        for (let y = 0; y < navLink.length; y++) {
            mainNavigationLink[y].style.color = '#FFFFFF';
        }
        logo.style.display='block';
        logoScroll.style.display='none';
    }
  });


  const openMenu = function () {
    littleDisplayNav.classList.remove('visually-hidden');
    openMenuBtn.classList.add('visually-hidden');
    closeMenuBtn.classList.remove('visually-hidden');
  };

  const closeMenu = function () {
    littleDisplayNav.classList.add('visually-hidden');
    closeMenuBtn.classList.add('visually-hidden');
    openMenuBtn.classList.remove('visually-hidden');
  };

  openMenuBtn.addEventListener('click', function() {
    openMenu();
  });

  closeMenuBtn.addEventListener('click', function() {
    closeMenu();
  });
