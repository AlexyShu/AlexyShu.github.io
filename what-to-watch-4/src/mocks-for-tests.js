export const mockFilm = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
  previewUrl: `img/the-grand-budapest-hotel.jpg`,
  bigPosterUrl: `img/the-grand-budapest-hotel-bg.jpg`,
  backgroundColor: `#ffffff`,
  videoUrl: `https://some-link`,
  trailerUrl: `https://some-link`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  votes: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  runTime: `99`,
  genre: `Comedy`,
  releaseYear: 2014,
  isFavorite: false
};


export const mockFilms = [
  {
    id: 1,
    name: `The Grand Budapest Hotel`,
    posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
    previewUrl: `img/the-grand-budapest-hotel.jpg`,
    bigPosterUrl: `img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#ffffff`,
    videoUrl: `https://some-link`,
    trailerUrl: `https://some-link`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    votes: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: `99`,
    genre: `Comedy`,
    releaseYear: 2014,
    isFavorite: false
  },
  {
    id: 2,
    name: `The Grand Budapest Hotel`,
    posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
    previewUrl: `img/the-grand-budapest-hotel.jpg`,
    bigPosterUrl: `img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#ffffff`,
    videoUrl: `https://some-link`,
    trailerUrl: `https://some-link`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    votes: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: `99`,
    genre: `Drama`,
    releaseYear: 2014,
    isFavorite: false
  },
  {
    id: 3,
    name: `The Grand Budapest Hotel`,
    posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
    previewUrl: `img/the-grand-budapest-hotel.jpg`,
    bigPosterUrl: `img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#ffffff`,
    videoUrl: `https://some-link`,
    trailerUrl: `https://some-link`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    votes: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: `99`,
    genre: `Action`,
    releaseYear: 2014,
    isFavorite: false
  }
];

export const mokcFunction = () => {};

export const MOCK_FILMS_COUNT = 8;

export const MOCK_STRING = `blablabla`;

export const MOCK_ACTIVE_FILTER = `All genres`;

export const MOCK_BOOL = false;

export const mockComments = [{
  "id": 1,
  "user": {
    "id": 4,
    "name": `Kate Muir`
  },
  "rating": 8.9,
  "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  "date": `2019-05-08T14:13:56.569Z`
}];

