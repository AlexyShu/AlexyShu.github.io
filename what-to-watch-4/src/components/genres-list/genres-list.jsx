import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ALL_GENRES, GENRES_COUNT, ARRAY_START} from "../../constants.js";
import {ActionCreator} from "../../reducer/state/state.js";
import {getFilmsByGenre} from "../../reducer/data/selectors.js";
import {getActiveFilter} from "../../reducer/state/selectors.js";
import FilmsList from "../films-list/films-list.jsx";
import withActiveItem from "../../hocs/with-active-item/with-active-item.jsx";

const FilmsWrapper = withActiveItem(FilmsList);

const GenresList = (props) => {
  const {films, onFilterClick, activeFilter, handleChange, filmsCount, filteredFilms, showMoreFilms} = props;
  const filters = [ALL_GENRES, ...new Set(films.map((film) => film.genre))];
  return (
    <React.Fragment>
      <ul className="catalog__genres-list">
        {filters.map((filter, i) => (
          <li
            onClick={()=> {
              handleChange(filter);
              onFilterClick(filter);
            }}
            key={filter + i}
            className={
              `catalog__genres-item ${filter === activeFilter ? `catalog__genres-item--active` : ``}`}
          >
            <a href="#" className="catalog__genres-link">
              {filter}
            </a>
          </li>)
        ).slice(ARRAY_START, GENRES_COUNT)}
      </ul>
      <FilmsWrapper
        filteredFilms = {filteredFilms}
        filmsCount = {filmsCount}
        showMoreFilms = {showMoreFilms}
      />
    </React.Fragment>
  );
};

GenresList.propTypes = {
  films: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        posterUrl: PropTypes.string,
        previewUrl: PropTypes.string,
        bigPosterUrl: PropTypes.string,
        backgroundColor: PropTypes.string,
        description: PropTypes.string,
        rating: PropTypes.number,
        votes: PropTypes.number,
        director: PropTypes.string,
        starring: PropTypes.arrayOf(PropTypes.string),
        runTime: PropTypes.string,
        genre: PropTypes.string,
        releaseYear: PropTypes.number,
        id: PropTypes.number,
        isFavorite: PropTypes.bool,
        videoUrl: PropTypes.string,
        trailerUrl: PropTypes.string
      })
  ).isRequired,
  onFilterClick: PropTypes.func.isRequired,
  activeFilter: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  filmsCount: PropTypes.number.isRequired,
  filteredFilms: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        posterUrl: PropTypes.string,
        previewUrl: PropTypes.string,
        bigPosterUrl: PropTypes.string,
        backgroundColor: PropTypes.string,
        description: PropTypes.string,
        rating: PropTypes.number,
        votes: PropTypes.number,
        director: PropTypes.string,
        starring: PropTypes.arrayOf(PropTypes.string),
        runTime: PropTypes.string,
        genre: PropTypes.string,
        releaseYear: PropTypes.number,
        id: PropTypes.number,
        isFavorite: PropTypes.bool,
        videoUrl: PropTypes.string,
        trailerUrl: PropTypes.string
      })
  ).isRequired,
  showMoreFilms: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  activeFilter: getActiveFilter(state),
  filteredFilms: getFilmsByGenre(state)
});

const mapDispatchToProps = (dispatch) => ({
  onFilterClick(genre) {
    dispatch(ActionCreator.changeGenre(genre));
  }
});

export {GenresList};
export default connect(mapStateToProps, mapDispatchToProps)(GenresList);
