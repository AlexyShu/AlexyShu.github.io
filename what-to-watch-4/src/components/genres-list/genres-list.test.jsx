import React from "react";
import renderer from "react-test-renderer";
import {GenresList} from "./genres-list.jsx";
import {Provider} from "react-redux";
import Namespace from "../../reducer/namespace.js";
import configureStore from "redux-mock-store";
import {mockFilms, mokcFunction, MOCK_ACTIVE_FILTER, MOCK_FILMS_COUNT} from "../../mocks-for-tests.js";
import {MemoryRouter} from "react-router-dom";

const mockStore = configureStore([]);

describe(`Render correct GenresList`, () => {
  it(`Render GenresList`, () => {
    const store = mockStore({
      [Namespace.DATA]: {
        films: mockFilms,
      },
      [Namespace.STATE]: {
        filmsCount: MOCK_FILMS_COUNT
      }
    });
    const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <GenresList
              films = {mockFilms}
              onFilterClick = {mokcFunction}
              activeFilter = {MOCK_ACTIVE_FILTER}
              handleChange= {mokcFunction}
              filmsCount = {MOCK_FILMS_COUNT}
              filteredFilms = {mockFilms}
              showMoreFilms = {mokcFunction}
            />
          </MemoryRouter>
        </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
