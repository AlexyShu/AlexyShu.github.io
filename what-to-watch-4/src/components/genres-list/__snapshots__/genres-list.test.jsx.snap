// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correct GenresList Render GenresList 1`] = `
Array [
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item "
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Comedy
      </a>
    </li>
    <li
      className="catalog__genres-item "
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Drama
      </a>
    </li>
    <li
      className="catalog__genres-item "
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        Action
      </a>
    </li>
  </ul>,
  <div
    className="catalog__movies-list"
  >
    <a
      className="small-movie-card catalog__movies-card small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      <article
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="175"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          The Grand Budapest Hotel
        </h3>
      </article>
    </a>
    <a
      className="small-movie-card catalog__movies-card small-movie-card__link"
      href="/films/2"
      onClick={[Function]}
    >
      <article
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="175"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          The Grand Budapest Hotel
        </h3>
      </article>
    </a>
    <a
      className="small-movie-card catalog__movies-card small-movie-card__link"
      href="/films/3"
      onClick={[Function]}
    >
      <article
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="175"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          The Grand Budapest Hotel
        </h3>
      </article>
    </a>
  </div>,
]
`;
